tests/unit/backend/utils/field-validation/email-validation.spec.ts

```diff
@@ -43,14 +57,19 @@ describe('Email field validation', () => {
       required: true,
       disabled: false,
     }
-    const response: ISingleAnswerResponse = {
+    const response = {
       _id: 'abc123',
       fieldType: BasicField.Email,
       question: 'random',
       answer: 'abc@163.com',
       isVisible: true,
-    }
-    const validateResult = validateField('formId', formField, response)
+      signature: 'some signature',
```

Comment:
I'm not sure this works to test signature validation unless the field has `isVerifiable: true`

Reply:
Actually the tests are not incorrect I think - I had added signatures to all the test cases, as the validator checks the signature when `isVerifiable: true`, but does not reject cases where `isVerifiable: false` but a signature is present. But I think this made the tests confusing to read so I've cleaned up and removed the signature from cases where `isVerifiable: false`.

I also added two more test cases for missing signature and incorrect signature

Finally I added docs for all the field validators

Reply:
To elaborate, the reason why I added signatures to the test in the first place, is because I shifted from mocking the `EmailValidator._isSignatureValid` method (i.e. mock the existence of a signature), to mocking the sdk (i.e. mock the validity of the signature)